lib/modules/manager/fleet/extract.ts

```diff
@@ -0,0 +1,109 @@
+import is from '@sindresorhus/is';
+import { loadAll } from 'js-yaml';
+import { regEx } from '../../../util/regex';
+import { GitTagsDatasource } from '../../datasource/git-tags';
+import { HelmDatasource } from '../../datasource/helm';
+import { checkIfStringIsPath } from '../terraform/util';
+import type { PackageDependency, PackageFile } from '../types';
+import type { FleetFile, FleetFileHelm, GitRepo } from './types';
+
+function extractGitRepo(doc: GitRepo): PackageDependency {
+  const dep: PackageDependency = {
+    depType: 'git_repo',
+    datasource: GitTagsDatasource.id,
+  };
+
+  const repo = doc.spec?.repo;
+  if (!repo) {
+    return {
+      ...dep,
+      skipReason: 'missing-depname',
+    };
+  }
+  dep.sourceUrl = repo;
+  dep.depName = repo;
+
+  const currentValue = doc.spec.revision;
+  if (!currentValue) {
+    return {
+      ...dep,
+      skipReason: 'no-version',
+    };
+  }
+
+  return {
+    ...dep,
+    currentValue,
+  };
+}
+
+function extractFleetFile(doc: FleetFileHelm): PackageDependency {
+  const dep: PackageDependency = {
+    depType: 'fleet',
+    datasource: HelmDatasource.id,
+  };
+
+  if (!doc.chart) {
+    return {
+      ...dep,
+      skipReason: 'missing-depname',
+    };
+  }
+  dep.depName = doc.chart;
+
+  if (!doc.repo) {
+    if (checkIfStringIsPath(doc.chart)) {
+      return {
+        ...dep,
+        skipReason: 'local-chart',
+      };
+    }
+    return {
+      ...dep,
+      skipReason: 'no-repository',
+    };
+  }
+  dep.registryUrls = [doc.repo];
+
+  const currentValue = doc.version;
+  if (!doc.version) {
+    return {
+      ...dep,
+      skipReason: 'no-version',
+    };
+  }
+
+  return {
+    ...dep,
+    currentValue,
+  };
+}
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string
+): PackageFile | null {
+  if (!content) {
+    return null;
+  }
+  const deps: PackageDependency[] = [];
+
+  if (regEx('fleet.ya?ml').test(packageFile)) {
```

Comment:
add a `try...catch` for invalid yaml parser errors, log debug which file failed and return `null`